src/components/transactions/TxDetails/index.tsx

```diff
@@ -159,21 +158,22 @@ const TxDetails = ({
   const chainId = useChainId()
   const { safe } = useSafeInfo()
 
-  const [pollCount] = useIntervalCounter(POLLING_INTERVAL)
-  const swapPollCount = isOpenSwapOrder(txSummary.txInfo) ? pollCount : 0
-
-  const [txDetailsData, error, loading] = useAsync<TransactionDetails>(
-    async () => {
-      if (txDetails && swapPollCount <= 0) {
-        return txDetails
-      }
-      return getTransactionDetails(chainId, txSummary.id)
+  const {
+    data: txDetailsData,
+    error,
+    isLoading: loading,
+    refetch,
+  } = useGetTransactionDetailsQuery(
+    { chainId, txId: txSummary.id },
+    {
+      pollingInterval: isOpenSwapOrder(txSummary.txInfo) ? POLLING_INTERVAL : undefined,
```

Comment:
Previously if it was not a swap order and there were already `txDetails` we would just return the existing `txDetails` without fetching them again. Is this still covered with the new hook?

Reply:
good point. You are right the new code is not equivalent and we actually aways do the the network request for it. I'll try to come up with a variant that has the old behavior.

Reply:
With @usame-algan we agreed that this is actually no longer necessary. I've written a detailed explanation why here: https://github.com/safe-global/safe-wallet-web/issues/4056